sdk/src/azure/iot/az_iot_common.c

```diff
@@ -27,6 +27,7 @@ AZ_NODISCARD az_result az_iot_message_properties_init(
   _az_PRECONDITION_NOT_NULL(properties);
   _az_PRECONDITION_VALID_SPAN(buffer, 0, true);
```

Comment:
Hmm, another way to think about the fix is to do this:
```suggestion
  _az_PRECONDITION_VALID_SPAN(buffer, written_length, true);
```

So, maybe first check the `_az_PRECONDITION(written_length >= 0);` and then ensure the minimum size of the span is  equal to written_length.

Reply:
Actually, you wouldn't even need to check `_az_PRECONDITION(written_length >= 0);`

`_az_PRECONDITION_VALID_SPAN` does it for you.